src-docs/src/assets/version-selector.js

```diff
@@ -0,0 +1,232 @@
+        position: absolute;
+        right: .5em;
+        top: .6em;
+    }
+    
+    #dropdown {
+        position: absolute;
+        min-width: calc(100% - 2px);
+        top: 100%;
+        left: 0;
+        box-shadow: 0 1px 3px rgba(0, 0, 0, 0.25), 0 4px 10px rgba(0, 0, 0, 0.12);
+        
+        margin: -5px 1px 0 1px;
+        padding-top: 5px;
+        white-space: nowrap;
+        border-radius: 0 0 4px 4px;
+        
+        background: #fff;
+        z-index: 1;
+    }
+    
+    :host(:not([aria-expanded="true"])) #dropdown {
+        display: none;
+    }
+    
+    #spacer {
+        appearance: none;
+        visibility: hidden;
+        pointer-events: none;
+        height: 0;
+        margin: 0 1px;
+        overflow: hidden;
+    }
+    
+    #spacer > a,
+    #dropdown > a {
+        display: block;
+        white-space: nowrap;
+        padding: 0.3em calc(3em - 1px) 0.3em calc(1em - 1px);
+        border-bottom: 1px solid #eee;
+        text-decoration: none;
+        color: var(--link-color);
+        position: relative;
+    }
+    
+    #dropdown > a:last-child {
+        border: 0;
+    }
+    
+    #dropdown > a:hover {
+        background: #efefef;
+    }
+    
+    a.latest:after {
+        content: "LATEST";
+        position: absolute;
+        right: .4rem;
+        font-size: 0.6em;
+        font-weight: 700;
+        top: 50%;
+        transform: translateY(-50%);
+        color: #999;
+    }
+    </style>
+    <a id="root" role="button" aria-labelledby="selected" aria-controls="dropdown" tabindex="0">
+      <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" role="img" 
+       width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24">
+       <g fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
+      <path d="M6 9l6 6l6-6"/></g>
+      </svg>
+      <span id="selected"></span>
+    </a>
+    <div id="dropdown" role="navigation"></div>
+    <div id="spacer" aria-hidden="true"></div>
+`;
+
+export class VersionSelector extends HTMLElement {
+  static get observedAttributes() {
+    return ['selected'];
+  }
+
+  constructor() {
+    super();
+    this.DOC_VERSIONS = [];
+    this.DOC_VERSION_LATEST = '';
+    this.attachShadow({ mode: 'open', delegatesFocus: true });
+
+    this._onBlur = ((e) => {
+      this._expand(false);
+      this.removeEventListener('blur', this._onBlur);
+    }).bind(this);
+  }
+
+  async fetchVersionJson() {
+    const url = 'https://d2p80mijmsa0ui.cloudfront.net/versions.json';
+    var branches = [];
+    try{
+      const response = await fetch(url);
+      if (response.ok) {
+      branches = await response.json();
+      branches.map((branch)=> this.DOC_VERSIONS.push(branch)).sort()
+      }
+    }
+    catch (err) {
+      console.warn(err)
+    }
+    const isLocalDev = window.location.host.includes('803'); // check if user running server locally, 8030 is default port
```

Comment:
nit: It would be great if we can check port in the url instead of checking whether `window.location.host` includes `803`.

Reply:
this is for testing while local. no major change!